File: lua/1160.lua
local function calculateYears(pa, pb, g1, g2)
    local anos = 0
    while anos <= 100 and pa <= pb do
        anos = anos + 1
        pa = pa + math.floor(pa * (g1 / 100))
        pb = pb + math.floor(pb * (g2 / 100))
    end

    if anos > 100 then
        return "Mais de 1 seculo."
    else
        return anos .. " anos."
    end
end

local n = io.read("*n")

for i = 1, n do
    local pa, pb, g1, g2 = io.read("*n", "*n", "*n", "*n")
    local result = calculateYears(pa, pb, g1, g2)
    print(result)
end

File: go/1160.go
package main

import (
	"fmt"
	"math"
)

func main() {
	var n, anos int
	var pa, pb, g1, g2 float64

	fmt.Scanf("%d", &n)

	for i := 0; i < n; i++ {
		fmt.Scanf("%f %f %f %f", &pa, &pb, &g1, &g2)

		anos = 0
		for {
			if anos > 100 || pa > pb {
				break
			}

			anos += 1
			pa = pa + math.Trunc(pa*(g1/100))
			pb = pb + math.Trunc(pb*(g2/100))
		}

		if anos > 100 {
			fmt.Println("Mais de 1 seculo.")
			continue
		}

		fmt.Printf("%d anos.\n", anos)

	}
}
